frontend/javascripts/viewer/model/volumetracing/section_labeling.ts

```diff
@@ -554,41 +555,196 @@ class VolumeLayer {
 
     const cx = sumCx / 6 / area;
     const cy = sumCy / 6 / area;
-    const zoomedPosition = this.get3DCoordinate([cx, cy]);
-    const pos = zoomedPositionToGlobalPosition(zoomedPosition, this.activeMag);
+    const outZoomedPosition: Vector3 = [0, 0, 0];
+    this.fast3DCoordinateFunction(cx, cy, outZoomedPosition);
+    const pos = zoomedPositionToGlobalPosition(outZoomedPosition, this.activeMag);
     return pos;
   }
+
+  getPlane(): OrthoView {
+    return this.plane;
+  }
+}
+
+const CANONICAL_BASES = {
+  [OrthoViews.PLANE_XY]: {
+    u: new THREE.Vector3(1, 0, 0),
+    v: new THREE.Vector3(0, 1, 0),
+    n: new THREE.Vector3(0, 0, 1),
+  },
+  [OrthoViews.PLANE_YZ]: {
+    u: new THREE.Vector3(0, 1, 0),
+    v: new THREE.Vector3(0, 0, 1),
+    n: new THREE.Vector3(1, 0, 0),
+  },
+  [OrthoViews.PLANE_XZ]: {
+    u: new THREE.Vector3(1, 0, 0),
+    v: new THREE.Vector3(0, 0, 1),
+    n: new THREE.Vector3(0, -1, 0),
+  },
+};
+const CANONICAL_NORMALS = {
+  [OrthoViews.PLANE_XY]: new THREE.Vector3(0, 0, 1),
+  [OrthoViews.PLANE_YZ]: new THREE.Vector3(1, 0, 0),
+  [OrthoViews.PLANE_XZ]: new THREE.Vector3(0, 1, 0),
+};
+
+export function mapTransformedPlane(
+  originalPlane: OrthoView,
+  transform: Transform,
+): [OrthoView, boolean /* swapped */, (scale: Vector3) => Vector2 /* adaptScaleFn */] {
+  if (originalPlane === "TDView") {
+    throw new Error("Unexpected 3D view");
+  }
+
+  const basis = CANONICAL_BASES[originalPlane];
+
+  const m = new THREE.Matrix4(
+    // @ts-ignore
+    ...invertAndTranspose(transform.affineMatrix),
+  );
+
+  // transform each basis vector
+  const u2 = basis.u.clone().applyMatrix4(m).normalize();
+  const n2 = basis.n.clone().applyMatrix4(m).normalize();
+
+  // find which canonical plane the transformed normal aligns with
+
+  let bestView: OrthoView = OrthoViews.PLANE_XY;
+  let bestDot = Number.NEGATIVE_INFINITY;
+
+  for (const [view, normal] of Object.entries(CANONICAL_NORMALS)) {
+    const dot = Math.abs(n2.dot(normal as THREE.Vector3));
+    if (dot > bestDot) {
+      bestDot = dot;
+      bestView = view as OrthoView;
+    }
+  }
+
+  const swapped = basis.u.dot(u2) === 0;
+
+  const adaptScaleFn = (scale: Vector3): Vector2 => {
+    const transposed = Dimensions.transDim(scale, originalPlane);
+    if (swapped) {
+      return [transposed[1], transposed[0]];
+    } else {
+      return [transposed[0], transposed[1]];
+    }
+  };
+
+  return [bestView, bestView === originalPlane && swapped, adaptScaleFn];
+}
+
+export function mapTransformedPlane2(
```

Comment:
And this is like an identity transform? Plane is goes out as a tuple attached with true? I am a little confused. I cannot follow this functions purpose

Reply:
sorry, it only existed for testing purposes. i removed it now.